fastai/optimizer.py

```diff
@@ -11,12 +11,24 @@
 # Cell
 class _BaseOptimizer():
     "Common functionality between `Optimizer` and `OptimWrapper`"
-    def all_params(self, n=slice(None), with_grad=False):
+    def all_params(self,
+        n=slice(None), # Where to slice the model or all as default
+        with_grad=False # Get all param tuples. If `True` select only those with a gradient
+    ):
         res = L((p,pg,self.state[p],hyper) for pg,hyper in zip(self.param_lists[n],self.hypers[n]) for p in pg)
         return L(o for o in res if hasattr(o[0], 'grad') and o[0].grad is not None) if with_grad else res
 
-    def _set_require_grad(self, rg, p,pg,state,h): p.requires_grad_(rg or state.get('force_train', False))
-    def freeze_to(self, n):
+    def _set_require_grad(self,
```

Comment:
Missing docstring and some type annotations.

Reply:
I dont thing we need a comment here, maybe will be enough if `n:int`???

Reply:
Yeah never mind it doesn't need a comment.